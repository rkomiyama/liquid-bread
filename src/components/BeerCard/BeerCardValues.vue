<template>
  <v-layout>
    <v-flex xs6>
      <h3>ABV</h3>
      <h3>IBU</h3>
      <h3>EBC</h3>
      <h3>SRM</h3>
      <div>First brewed</div>
      <div>Volume</div>
      <div>Attenuation level</div>
      <div>pH</div>
      <div>Boil volume</div>
      <div>Target OG</div>
      <div>Target FG</div>
      <div>Contributed by</div>
    </v-flex>
    <v-flex xs6>
      <h3>{{ beer.abv ? beer.abv : 'N/A' }}%</h3>
      <h3>{{ beer.ibu ? beer.ibu : 'N/A' }} BU</h3>
      <h3>{{ beer.ebc ? beer.ebc : 'N/A' }}</h3>
      <h3>{{ beer.srm ? beer.srm : 'N/A' }}</h3>
      <div>{{ beer.first_brewed ? beer.first_brewed : 'N/A' }}</div>
      <div>{{ beer.volume ? `${beer.volume.value} ${beer.volume.unit}` : 'N/A' }}</div>
      <div>{{ beer.attenuation_level ? beer.attenuation_level : 'N/A' }}</div>
      <div>{{ beer.ph ? beer.ph : 'N/A' }}</div>
      <div>{{ beer.boil_volume ? `${beer.boil_volume.value} ${beer.boil_volume.unit}` : 'N/A' }}</div>
      <div>{{ beer.target_og ? beer.target_og : 'N/A' }}</div>
      <div>{{ beer.target_fg ? beer.target_fg : 'N/A' }}</div>
      <div>{{ beer.contributed_by ? beer.contributed_by : 'N/A' }}</div>
    </v-flex>
  </v-layout>
</template>

<script>
export default {
  name: "BeerCardValues",
  props: {
    beer: {
      type: Object,
      default: () => {}
    }
  }
};
</script>

<style lang="scss" scoped>
</style>
