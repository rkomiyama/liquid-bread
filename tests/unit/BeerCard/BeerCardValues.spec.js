import Vue from "vue";
import Vuetify from "vuetify";
import { shallowMount } from "@vue/test-utils";
import { createRenderer } from "vue-server-renderer";
import BeerCardValues from "@/components/BeerCard/BeerCardValues.vue";
import beerData from "#/mockData/beerData.json";

Vue.use(Vuetify);

let wrapper;

describe("BeerCardValues.vue", () => {
  beforeEach(() => {
    wrapper = shallowMount(BeerCardValues, {
      propsData: { beer: beerData }
    });
  });

  it("BeerCardValues has same HTML structure", () => {
    const renderer = createRenderer();
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err);
      expect(str).toMatchSnapshot();
    });
  });
});
