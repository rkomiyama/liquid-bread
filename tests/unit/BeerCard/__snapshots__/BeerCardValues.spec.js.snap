// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeerCardValues.vue BeerCardValues has same HTML structure 1`] = `
<v-layout-stub tag="div">
  <v-flex-stub tag="div" xs6="">
    <h3>ABV</h3>
    <h3>IBU</h3>
    <h3>EBC</h3>
    <h3>SRM</h3>
    <div>First brewed</div>
    <div>Volume</div>
    <div>Attenuation level</div>
    <div>pH</div>
    <div>Boil volume</div>
    <div>Target OG</div>
    <div>Target FG</div>
    <div>Contributed by</div>
  </v-flex-stub>
  <v-flex-stub tag="div" xs6="">
    <h3>4.5%</h3>
    <h3>60 BU</h3>
    <h3>20</h3>
    <h3>10</h3>
    <div>09/2007</div>
    <div>20 liters</div>
    <div>75</div>
    <div>4.4</div>
    <div>25 liters</div>
    <div>1044</div>
    <div>1010</div>
    <div>Sam Mason &lt;samjbmason&gt;</div>
  </v-flex-stub>
</v-layout-stub>
`;
